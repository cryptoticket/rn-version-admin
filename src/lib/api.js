import axios from 'axios';

// if user auth token exists then assign it to auth header
const userAuthToken = localStorage.getItem('token');
if(userAuthToken) axios.defaults.headers.common['Authorization'] = `Bearer ${userAuthToken}`;

/**
 * Bundles API
 */

/**
 * Return bundles by page number 
 */
async function getBundles(page = 0) {
	return await axios.get('/api/v1/bundles', { params: {
		page: page
	}});
}

/**
 * Users API
 */

/**
 * Return users by page number 
 */
async function getUsers(page = 0) {
	return await axios.get('/api/v1/users', { params: {
		page: page
	}});
}

export default {
	// bundles API
	getBundles,
	// users API
	getUsers
};
