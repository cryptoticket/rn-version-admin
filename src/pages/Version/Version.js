import { Button } from 'primereact/button';
import { Card } from 'primereact/card';
import { Column } from 'primereact/column';
import { DataTable } from 'primereact/datatable';
import { Dialog } from 'primereact/dialog';
import { Dropdown } from 'primereact/dropdown';
import  { InputText } from 'primereact/inputtext';
import { Paginator } from 'primereact/paginator';
import React from 'reactn';

import { api } from '../../lib';

/**
 * Component renders a page with versions
 */
export default class Version extends React.Component {
	// default state
	state = {
		bundles: [],
		isDeleteDialogVisible: false,
		isDialogVisible: false,
		latestBundleAndroid: {},
		latestBundleIos: {},
		paginationCurrentPage: 0,
		paginationItemsPerPage: 0,
		paginationTotalItemsCount: 0,
		selectedBundle: null
	};

	/**
	 * On component init
	 */
	async componentDidMount() {
		try {
			this.setGlobal({isLoading: true});
			const respBundles = await api.getBundles(this.state.paginationCurrentPage);
			const respLatestBundleAndroid = await api.getLatestBundle('android');
			const respLatestBundleIos = await api.getLatestBundle('ios');
			this.setState({
				bundles: respBundles.data,
				latestBundleAndroid: respLatestBundleAndroid.data,
				latestBundleIos: respLatestBundleIos.data,
				paginationItemsPerPage: +respBundles.headers['x-limit'],
				paginationTotalItemsCount: +respBundles.headers['x-total-count']
			});
		} catch(err) {
			console.error(err);
			this.global.growl.show({severity: 'error', summary: 'Error', detail: 'Error on loading bundles'});
		} finally {
			this.setGlobal({isLoading: false});
		}
	};

	/**
	 * Returns delete dialog footer
	 * @return {Object} JSX template
	 */
	getDeleteDialogFooter = () => {
		return (<div>
			<Button label="Delete" icon="pi pi-trash" onClick={this.onDelete} className="p-button-danger" />
			<Button label="Cancel" icon="pi pi-times" onClick={() => this.setState({isDeleteDialogVisible: false})} className="p-button-secondary" />
		</div>);
	};

	/**
	 * Returns dialog footer
	 * @return {Object} JSX template
	 */
	getDialogFooter = () => {
		return (<div>
			<Button label="Save" icon="pi pi-check" onClick={this.onSaveClick} />
			<Button label="Cancel" icon="pi pi-times" onClick={() => this.setState({isDialogVisible: false})} className="p-button-secondary" />
		</div>);
	};

	/**
	 * On bundle delete
	 */
	onDelete = async () => {
		try {
			await api.deleteBundle(this.state.selectedBundle._id);
			this.setState({
				isDeleteDialogVisible: false,
				selectedBundle: null
			});
			await this.componentDidMount();
			this.global.growl.show({severity: 'success', summary: 'Success', detail: 'Bundle deleted'});
		} catch(err) {
			console.error(err);
			this.global.growl.show({severity: 'error', summary: 'Error', detail: 'Error on deleting bundle'});
		}
	};

	/**
	 * On "delete" button click
	 * @param {Object} rowData model data
	 */
	onDeleteClick = (rowData) => {
		this.setState({
			isDeleteDialogVisible: true,
			selectedBundle: rowData
		});
	};

	/**
	 * On "edit" button press
	 * @param {Object} rowData model data
	 */
	onEditClick = (rowData) => {
		this.setState({
			isDialogVisible: true,
			selectedBundle: rowData
		});
	};

	/**
	 * On "save" button click in dialog
	 */
	onSaveClick = async () => {
		try {
			await api.updateBundle(this.state.selectedBundle);
			this.setState({
				isDialogVisible: false,
				selectedBundle: null
			});
			await this.componentDidMount();
			this.global.growl.show({severity: 'success', summary: 'Success', detail: 'Bundle updated'});
		} catch(err) {
			console.error(err);
			this.global.growl.show({severity: 'error', summary: 'Error', detail: 'Error on updating bundle'});
		}
	};

	/**
	 * Renders JSX template
	 * @return {Object} JSX template
	 */
	render() {
		return (
			<div>
				<Card>
					<span>Android latest version: <strong>{Object.keys(this.state.latestBundleAndroid).length > 0 ? this.state.latestBundleAndroid.version : 'not available'}</strong></span>
				</Card>
				<Card>
					<span>iOS latest version: <strong>{Object.keys(this.state.latestBundleIos).length > 0 ? this.state.latestBundleIos.version : 'not available'}</strong></span>
				</Card>
				{/* versions table */}
				<Card>
					<h3>Versions</h3>
					<DataTable value={this.state.bundles}>
						<Column field="platform" header="Platform" />
						<Column field="version" header="Version" />
						<Column field="is_update_required" header="Is update required" body={(rowData) => String(rowData['is_update_required'])} />
						<Column field="storage" header="Storage type" />
						<Column field="desc" header="Description" />
						<Column field="created_at" header="Created at" />
						<Column field="updated_at" header="Updated at" />
						<Column body={(rowData) => (
							<React.Fragment>
								<span>
									<Button className="p-button-warning" label="Edit" icon="pi pi-pencil" onClick={() => this.onEditClick(rowData)} />
								</span>
								<span className="ml-5">
									<Button className="p-button-danger" label="Delete" icon="pi pi-trash" onClick={() => this.onDeleteClick(rowData)} />
								</span>
							</React.Fragment>
						)} />
					</DataTable>
					<Paginator 
						rows={this.state.paginationItemsPerPage} 
						totalRecords={this.state.paginationTotalItemsCount} 
						first={this.state.paginationItemsPerPage * this.state.paginationCurrentPage}
						onPageChange={(e) => this.setState({paginationCurrentPage: e.first / this.state.paginationItemsPerPage}, () => this.componentDidMount())} />
				</Card>
				{/* bundle dialog window */}
				{this.state.selectedBundle &&
					<Dialog header={`Updating ${this.state.selectedBundle.platform} bundle for version ${this.state.selectedBundle.version}`} style={{width: '30%'}} visible={this.state.isDialogVisible} footer={this.getDialogFooter()} onHide={() => this.setState({isDialogVisible: false})}>
						<div>
							<div>
								<div>
									Is update required
								</div>
								<div>
									<Dropdown 
										value={this.state.selectedBundle.is_update_required} 
										options={[{label: 'Yes', value: true}, {label: 'No', value: false}]} 
										onChange={(e) => this.setState({selectedBundle: {...this.state.selectedBundle, ...{is_update_required: e.target.value}}})} 
									/>
								</div>
							</div>
							<div>
								<div>
									Description
								</div>
								<div>
									<InputText 
										value={this.state.selectedBundle.desc} 
										onChange={(e) => this.setState({selectedBundle: {...this.state.selectedBundle, ...{desc: e.target.value}}})} 
									/>
								</div>
							</div>
						</div>
					</Dialog>
				}
				{/* delete dialog */}
				{this.state.isDeleteDialogVisible &&
					<Dialog header={'Deleting bundle'} style={{width: '30%'}} visible={this.state.isDeleteDialogVisible} footer={this.getDeleteDialogFooter()} onHide={() => this.setState({isDeleteDialogVisible: false})}>
						<div>
							Delete {this.state.selectedBundle.platform} bundle with version {this.state.selectedBundle.version} ?
						</div>
					</Dialog>
				}
			</div>
		);
	}
}
