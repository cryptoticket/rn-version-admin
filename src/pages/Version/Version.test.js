import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'reactn';
import renderer from 'react-test-renderer';
import semver from 'semver';

import { api } from '../../lib';
import Version from './Version';

configure({ adapter: new Adapter() });

describe('Version component', () => {
    describe('componentDidMount()', () => {
		it('should load bundles', async () => {
			const bundles = [
				{
					platform: 'android',
					version: '1.0.0',
					is_update_required: false,
					storage: 'file',
					desc: 'desc',
					created_at: 1,
					updated_at: 2
				},
				{
					platform: 'ios',
					version: '1.0.0',
					is_update_required: false,
					storage: 'file',
					desc: 'desc',
					created_at: 1,
					updated_at: 2
				}
			];
			api.getBundles = jest.fn().mockReturnValue({
				headers: {
					'x-limit': 20,
					'x-total-count': 2
				},
				data: bundles
			});
			api.getLatestBundle = jest.fn((platform) => {
				return platform === 'android' ? { data: bundles[0] } : { data: bundles[1] };
			});
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setGlobal = jest.fn();
			await instance.componentDidMount();
			expect(instance.setGlobal).toHaveBeenCalledWith({isLoading: true});
			expect(api.getBundles).toHaveBeenCalled();
			expect(api.getLatestBundle).toHaveBeenCalledWith('android');
			expect(api.getLatestBundle).toHaveBeenCalledWith('ios');
			expect(instance.state.bundles).toEqual(bundles);
			expect(instance.state.latestBundleAndroid).toEqual(bundles[0]);
			expect(instance.state.latestBundleIos).toEqual(bundles[1]);
			expect(instance.state.paginationItemsPerPage).toEqual(20);
			expect(instance.state.paginationTotalItemsCount).toEqual(2);
			expect(instance.setGlobal).toHaveBeenCalledWith({isLoading: false});
		});

		it('should throw on api error', async () => {
			api.getBundles = () => {
				return Promise.reject('expected test error');
			};
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setGlobal({
				growl: {
					show: jest.fn()
				}
			});
			await instance.componentDidMount();
			expect(instance.global.growl.show).toHaveBeenCalledWith({severity: 'error', summary: 'Error', detail: 'Error on loading bundles'});
		});
	});

	describe('getDeleteDialogFooter()', () => {
		it('should render delete dialog footer', function () {
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			const component = renderer.create(instance.getDeleteDialogFooter());
			expect(component.toJSON()).toMatchSnapshot();
		});
	});

	describe('getDialogFooter()', () => {
		it('should render user dialog footer', function () {
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			const component = renderer.create(instance.getDialogFooter());
			expect(component.toJSON()).toMatchSnapshot();
		});
	});

	describe('onDelete()', () => {
		it('should delete bundle', async () => {
			api.deleteBundle = jest.fn();
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setState({
				selectedBundle: {
					_id: 1
				}
			});
			instance.componentDidMount = jest.fn();
			instance.setGlobal({
				growl: {
					show: jest.fn()
				}
			});
			await instance.onDelete();
			expect(api.deleteBundle).toHaveBeenCalled();
			expect(instance.state.isDeleteDialogVisible).toEqual(false);
			expect(instance.state.selectedBundle).toEqual(null);
			expect(instance.componentDidMount).toHaveBeenCalled();
			expect(instance.global.growl.show).toHaveBeenCalledWith({severity: 'success', summary: 'Success', detail: 'Bundle deleted'});
		});

		it('should throw on delete bundle error', async () => {
			api.deleteBundle = () => {
				return Promise.reject('expected test error');
			};
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setState({
				selectedBundle: {
					_id: 1
				}
			});
			instance.setGlobal({
				growl: {
					show: jest.fn()
				}
			});
			await instance.onDelete();
			expect(instance.global.growl.show).toHaveBeenCalledWith({severity: 'error', summary: 'Error', detail: 'Error on deleting bundle'});
		});
	});

	describe('onDeleteClick()', () => {
		it('should show delete dialog', () => {
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.onDeleteClick({id: 1});
			expect(instance.state.isDeleteDialogVisible).toEqual(true);
			expect(instance.state.selectedBundle).toEqual({id: 1});
		});
	});

	describe('onEditClick()', () => {
		it('should show edit dialog', () => {
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.validateEditForm = jest.fn();
			instance.onEditClick({id: 1});
			expect(instance.state.isDialogVisible).toEqual(true);
			expect(instance.state.selectedBundle).toEqual({id: 1});
			expect(instance.validateEditForm).toHaveBeenCalled();
		});
	});

	describe('onSaveClick()', () => {
		it('should update bundle', async () => {
			api.updateBundle = jest.fn();
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.componentDidMount = jest.fn();
			instance.setGlobal({
				growl: {
					show: jest.fn()
				}
			});
			await instance.onSaveClick();
			expect(api.updateBundle).toHaveBeenCalled();
			expect(instance.state.isDialogVisible).toEqual(false);
			expect(instance.state.selectedBundle).toEqual(null);
			expect(instance.componentDidMount).toHaveBeenCalled();
			expect(instance.global.growl.show).toHaveBeenCalledWith({severity: 'success', summary: 'Success', detail: 'Bundle updated'});
		});

		it('should throw on bundle update API error', async () => {
			api.updateBundle = () => {
				return Promise.reject('expected test error');
			};
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setGlobal({
				growl: {
					show: jest.fn()
				}
			});
			await instance.onSaveClick();
			expect(instance.global.growl.show).toHaveBeenCalledWith({severity: 'error', summary: 'Error', detail: 'Error on updating bundle'});
		});
	});

	describe('validateEditForm()', () => {
		it('should set edit form validity to false if apply_from_version is not in semver format', () => {
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setState({
				selectedBundle: {
					apply_from_version: 'INVALID'
				}
			});
			semver.valid = jest.fn();
			instance.validateEditForm();
			expect(semver.valid).toHaveBeenCalledWith('INVALID');
			expect(instance.state.isEditFormValid).toEqual(false);
		});

		it('should set edit form validity to false if apply_from_version is less than existing bundle version', () => {
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setState({
				selectedBundle: {
					apply_from_version: '1.0.0',
					version: '1.0.0'
				}
			});
			semver.valid = jest.fn(() => true); // NOTICE: this is a workaround as previous test's semver mock is somehow not reset
			semver.lt = jest.fn();
			instance.validateEditForm();
			expect(semver.lt).toHaveBeenCalledWith('1.0.0', '1.0.0');
			expect(instance.state.isEditFormValid).toEqual(false);
		});

		it('should set edit form validity to true if form is valid', () => {
			const wrapper = shallow(<Version />);
			const instance = wrapper.instance();
			instance.setState({
				selectedBundle: {
					apply_from_version: '1.0.0',
					version: '1.0.1'
				}
			});
			semver.valid = jest.fn(() => true); // NOTICE: this is a workaround as previous test's semver mock is somehow not reset
			semver.lt = jest.fn(() => true); // NOTICE: this is a workaround as previous test's semver mock is somehow not reset
			instance.validateEditForm();
			expect(instance.state.isEditFormValid).toEqual(true);
		});
	});

	describe('render()', () => {
		it('should render component', function () {
			const component = renderer.create(<Version />);
			expect(component.toJSON()).toMatchSnapshot();
		});
	});
});
