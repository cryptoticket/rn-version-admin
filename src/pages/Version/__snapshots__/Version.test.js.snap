// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Version component getDeleteDialogFooter() should render delete dialog footer 1`] = `
<div>
  <button
    className="p-button p-component p-button-danger p-button-text-icon-left"
    onClick={[Function]}
  >
    <span
      className="pi pi-trash p-c p-button-icon-left"
    />
    <span
      className="p-button-text p-c"
    >
      Delete
    </span>
  </button>
  <button
    className="p-button p-component p-button-secondary p-button-text-icon-left"
    onClick={[Function]}
  >
    <span
      className="pi pi-times p-c p-button-icon-left"
    />
    <span
      className="p-button-text p-c"
    >
      Cancel
    </span>
  </button>
</div>
`;

exports[`Version component getDialogFooter() should render user dialog footer 1`] = `
<div>
  <button
    className="p-button p-component p-button-text-icon-left p-disabled"
    disabled={true}
    onClick={[Function]}
  >
    <span
      className="pi pi-check p-c p-button-icon-left"
    />
    <span
      className="p-button-text p-c"
    >
      Save
    </span>
  </button>
  <button
    className="p-button p-component p-button-secondary p-button-text-icon-left"
    onClick={[Function]}
  >
    <span
      className="pi pi-times p-c p-button-icon-left"
    />
    <span
      className="p-button-text p-c"
    >
      Cancel
    </span>
  </button>
</div>
`;

exports[`Version component render() should render component 1`] = `
<div>
  <div
    className="p-card p-component"
    style={null}
  >
    <div
      className="p-card-body"
    >
      <div
        className="p-card-content"
      >
         
        <span>
          Android latest version: 
          <strong>
            not available
          </strong>
        </span>
         
      </div>
    </div>
  </div>
  <div
    className="p-card p-component"
    style={null}
  >
    <div
      className="p-card-body"
    >
      <div
        className="p-card-content"
      >
         
        <span>
          iOS latest version: 
          <strong>
            not available
          </strong>
        </span>
         
      </div>
    </div>
  </div>
  <div
    className="p-card p-component"
    style={null}
  >
    <div
      className="p-card-body"
    >
      <div
        className="p-card-content"
      >
         
        <h3>
          Versions
        </h3>
        <div
          className="p-datatable p-component"
          id={null}
          style={null}
        >
          <div
            className="p-datatable-wrapper"
          >
            <table
              className={null}
              role="grid"
              style={null}
            >
              <thead
                className="p-datatable-thead"
              >
                <tr>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Platform
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Version
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Apply from version
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Is update required
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Storage type
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Description
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Created at
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    >
                      Updated at
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className=""
                    colSpan={null}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onDragOver={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    rowSpan={null}
                    style={null}
                    tabIndex={-1}
                  >
                    <span
                      className="p-column-title"
                    />
                  </th>
                </tr>
              </thead>
              <tbody
                className="p-datatable-tbody"
              />
            </table>
          </div>
        </div>
        <div
          className="p-paginator p-component p-unselectable-text"
          style={null}
        >
          <button
            className="p-paginator-first p-paginator-element p-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="p-paginator-icon pi pi-step-backward"
            />
          </button>
          <button
            className="p-paginator-prev p-paginator-element p-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="p-paginator-icon pi pi-caret-left"
            />
          </button>
          <span
            className="p-paginator-pages"
          />
          <button
            className="p-paginator-next p-paginator-element p-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="p-paginator-icon pi pi-caret-right"
            />
          </button>
          <button
            className="p-paginator-last p-paginator-element p-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="p-paginator-icon pi pi-step-forward"
            />
          </button>
        </div>
         
      </div>
    </div>
  </div>
</div>
`;
